exercises/acronym/example.rs

```diff
@@ -0,0 +1,20 @@
+pub fn abbreviate(phrase: &str) -> String {
+    phrase.split(|c: char| c.is_whitespace() || !c.is_alphanumeric())
+          .flat_map(|word| split_camel(word))
+          .filter(|word| !word.is_empty())
```

Comment:
I wondered whether any element in the vec returned by `split_camel` can ever be empty, but I don't think it can, because that implies `word_start == i + 1`, and that can never be because `word_start` is only ever set to a previous iteration's `i + 1` (and previous iteration's `i` is less than this iteration's `i`) or zero. Can you come up with a situation where it would?

You may ask "is it a concern that I had to think about this", and maybe it isn't - knowing that no word is empty will prevent panics on the `split_at(1)`, so maybe this is to make abundantly clear that it won't panic. So I appreciate that it prevents the reader form having to worry about that.

For these reasons, I don't mind if this line stays, I leave it up to you.

If you want to remove it but are concerned that now the reader needs to expend effort to ensure that we don't have an empty word, I can suggest that:

* You only want the first character of each word
* https://stackoverflow.com/questions/37635978/how-do-i-access-the-first-character-of-a-string-in-rust tells us a way to do that that returns an `Option`, so the reader can be sure that the case is covered.